src/languages/fsharp.js

```diff
@@ -89,52 +78,317 @@ export default function(hljs) {
+    "load",
+    "time",
+    "help",
+    "quit"
+  ];
+
+  const LITERALS = [
+    "true",
+    "false",
+    "null",
+    "Some",
+    "None",
+    "Ok",
+    "Error",
+    "infinity",
+    "infinityf",
+    "nan",
+    "nanf",
+    "__LINE__",
+    "__SOURCE_DIRECTORY__",
+    "__SOURCE_FILE__"
+  ];
+
+  const TYPES = [
+    // basic types
+    "bool",
+    "byte",
+    "sbyte",
+    "int8",
+    "int16",
+    "int32",
+    "uint8",
+    "uint16",
+    "uint32",
+    "int",
+    "uint",
+    "int64",
+    "uint64",
+    "nativeint",
+    "unativeint",
+    "decimal",
+    "float",
+    "double",
+    "float32",
+    "single",
+    "char",
+    "string",
+    "unit",
+    "bigint",
+    // other native types or lowercase aliases
+    "option",
+    "voption",
+    "list",
+    "array",
+    "seq",
+    "byref",
+    "exn",
+    "inref",
+    "nativeptr",
+    "obj",
+    "outref",
+    "voidptr"
+  ];
+  
+  const BUILTINS = [
+    // Somewhat arbitrary list of builtin functions and values.
+    // Most of them are declared in Microsoft.FSharp.Core
+    // I tried to stay relevant by adding only the most idiomatic
+    // and most used symbols that are not already declared as types.
+    "not",
+    "ref",
+    "raise",
+    "reraise",
+    "dict",
+    "readOnlyDict",
+    "set",
+    "enum",
+    "sizeof",
+    "typeof",
+    "typedefof",
+    "nameof",
+    "nullArg",
+    "invalidArg",
+    "invalidOp",
+    "id",
+    "fst",
+    "snd",
+    "ignore",
+    "lock",
+    "using",
+    "box",
+    "unbox",
+    "tryUnbox",
+    "printf",
+    "printfn",
+    "sprintf",
+    "eprintf",
+    "eprintfn",
+    "fprintf",
+    "fprintfn",
+    "failwith",
+    "failwithf"
+  ];
+
+  // (* potentially multi-line ML style comment *)
+  const ML_COMMENT =
+    hljs.COMMENT(/\(\*(?!\))/, /\*\)/, {
+      contains: ["self"]
+    });
+  // Either a multi-line (* ML style comment *) or a single line // C style comment.
+  const FSHARP_COMMENT = {
+    variants: [
+      ML_COMMENT,
+      hljs.C_LINE_COMMENT_MODE,
+    ]
+  };
+
+  // All the following string definitions are potentially multi-line:
+  // "..."
+  const QUOTED_STRING = {
+    scope: 'string',
+    begin: '"',
+    end: '"',
+    contains: [
+      hljs.BACKSLASH_ESCAPE
+    ],
+    relevance: 0
+  };
+  // @"..."
+  const VERBATIM_STRING = {
+    scope: 'string',
+    begin: '@"',
+    end: '"',
+    contains: [
+      {
+        match: '""'
+      },
+      hljs.BACKSLASH_ESCAPE
+    ]
+  };
+  //"""..."""
+  const TRIPLE_QUOTED_STRING = {
+    scope: 'string',
+    begin: '"""',
+    end: '"""',
+    relevance: 2
+  };
+  const SUBST = {
+    scope: 'subst',
+    begin: /\{/,
+    end: /\}/
+  };
+  // $"...{1+1}..."
+  const INTERPOLATED_STRING = {
+    scope: 'string',
+    begin: /\$"/,
+    end: '"',
+    contains: [
+      {
+        match: /\{\{/
+      },
+      {
+        match: /\}\}/
+      },
+      hljs.BACKSLASH_ESCAPE,
+      SUBST
+    ]
+  };
+  // $@"...{1+1}..."
```

Comment:
Double braces or single braces? Your regex includes two...

Reply:
Placeholders are defined between single braces, but doubling a brace escapes it.
```fsharp
$"this is {{an example}} and here is an actual placeholder: {1+1}"
$@"this is {{an example}} and here is an actual placeholder: {1+1}"
$"""this is {{an example}} and here is an actual placeholder: {1+1}"""
// All the above are interpreted as "this is {an example} and here is an actual placeholder: 2"
```

Reply:
I'm wondering if this should be:

```
      {
        begin: /\{\{/
        end: /\}\}/
      },
```

?

Reply:
Mmh, I don't think so: an escaped brace does not imply another matching escaped brace (something like `$"a brace: {{. another brace: {{"` is entirely possible, and evaluated as `"a brace: {. another brace: {"`)

Reply:
Ah, I'll blame it on the fact that the examples made them appear paired.  I understand now. :-)